src/app-layer-parser.c

```diff
@@ -1438,6 +1441,32 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
         }
     }
 
+    if (pstate->flags & APP_LAYER_PARSER_LAYERED_PACKET) {
+        // hardcoded logic
+        if (alproto == ALPROTO_HTTP2) {
+            const uint8_t *b = NULL;
+            uint32_t b_len = 0;
+            for (uint32_t i = 0;; i++) {
+                if (!SCHttp2GetLayered(alstate, &b, &b_len, i)) {
+                    SCHttp2ClearLayered(alstate);
+                    break;
+                }
+                Packet *np = PacketPseudoFromFlow(f, f->protoctx, direction);
+                if (np == NULL) {
+                    // TODO log warning
+                    continue;
+                }
+                PKT_SET_SRC(np, PKT_SRC_APP_LAYER_LAYERED);
+                // TODO np->flags |= PKT_PSEUDO_DETECTLOG_FLUSH;
+                np->payload = SCMalloc(b_len);
+                memcpy(np->payload, b, b_len);
+                np->payload_len = b_len;
+                PacketEnqueueNoLock(&tv->decode_pq, np);
```

Comment:
⚠️ this is the second key step :

@victorjulien is `&tv->decode_pq` the right place to ensue such pseudo packets ?
I fear this will not work if I encapsulate my DNS over HTTP2 on GRE or other tunnels...

(`&fw->pq` is not already accessible here)